### PR TITLE
Radiation in flames

### PR DESCRIPTION
**Pull request by cannondale1492**
Hallo Ray,

i am not sure if i did this in the right way. I tried to implement the changes you suggested me in January for the use of radiation in counterflowflames. I am sorry if i did something wrong in simply creating this pull request with all my commits. Please tell if i should do that in another way.

Best regards

Andy

### CODE REVIEW

**Comment by speth** [comment]
Andy,

This looks quite good so far. I think submitting it as a pull request is the easiest way to see what changes are part of this, even though merging these changes eventually will need to be done via Google Code.

A few of improvements that I would like to see:
- A short description and reference for the model being implemented here
- An example that calls these functions (it could just be an additional case in one of the existing examples, e.g. mixture averaged / multicomponent / multi+radiation).
- A test (in the Python test suite) that exercises this code
- There should be no logging from calling `solveRadiationEqn` or `setBoundaryEmissivities`
- I might suggest renaming `solveRadiationEqn` to `enableRadiation` and the corresponding Python property to `radiation_enabled`, as this is adding a term to an equation, not adding a new equation to be solved
- There are a few minor formatting issues (tabs instead of spaces, trailing whitespace) that could be fixed.

**Comment by cannondale1492** [comment]
Hello Ray,

i wrote an example in the cython examples, but I'm not sure if it's the right place for it. Do you want a case in the Cxx examples too? Do you have a tip for me where i can place my test in the python suite (file and line)?

Best regards 
Andy

**Comment by speth** [comment]
Andy,

The modification to the cython example looks like a perfectly good place for that. There's no need to add a c++ example.

For the Python test, you can add it to `interfaces/cython/cantera/test/test_onedim.py`. I think you could follow the example of `TestDiffusionFlame.test_mixture_averaged` (starts on line 419). You'll need to generate and add a reference file for the regression test; there are instructions in the comment at the start of the class which should work.

I'm a little concerned about taking the values from the Cosilab manual, as it's a commercial product. I assume that the original source is something in the scientific literature, which it would be better to cite instead.

Regards,
Ray

**Comment by cannondale1492** [comment]
Hello Ray,

i have a short question: Is it okay/possible to use a second reference file (referenceFileRad) for my test case in the TestDiffusionFlame class, or should I use a whole new test class?
I will find out the original source for the polynomial coefficients.

Regards
Andy

**Comment by speth** [comment]
Yes, having this test method in class `TestDiffusionFlame` is just fine. You could just make `referenceFile` a local variable in each test method, since each reference file is only used in a single test method.

**Comment by cannondale1492** [comment]
Hello Ray,

could you please have a look at the test case? It should work, but i am not able create the referenceFile for it.

Best regards
Andy

**Comment by speth** [comment]
I can create the necessary reference file; that's not a problem.

File: src/oneD/StFlow.cpp [hunk 1]
```diff
@@ -280,6 +280,93 @@ void StFlow::eval(size_t jg, doublereal* xg,
...
+        // loop over all grid points
+        for (size_t jnew = 0; jnew < m_points; jnew++){
+            // helping variable for the calculation
+            doublereal radiative_heat_loss = 0;
+            // calculation of the mean Planck absorption coefficient
+            // initialization of the sums to zero
+            sum_H2O = 0;
+            sum_CO2 = 0;
+
+            // loop for the polynomial rows
```
Comment (speth) [inline src/oneD/StFlow.cpp]: Are these formulas (lines 332-341) for calculating k_P implemented correctly? Where do they come from? It's suspicious that k_P_H2O and k_P_CO2 are calculated in every iteration of the loop, but only the value after the last iteration is ever used. They don't seem to be in either of the references mentioned above. The Hubbard and Tien paper has a graph of k_P (Fig. 2), but it does not match my calculations using these formulas.

**Comment by cannondale1492** [comment]
Hello Ray,

I recalculated the polynomial lines and I think they fit to the data from Hubbard and Tien. Anyway it seems that the polynomial coefficients are only mentioned in the Cosilab Manual. I found this approach too:
http://www.sandia.gov/TNF/radiation.html
Could we use that?
Alternatively I found the coefficients from the Hitran and Hitemp databases in the book [Radiative Heat Transfer - Michael F. Modest](Figure 11-31 on page 368). I made a compilation for the different approaches and sent it to you via mail.

Best regards

Andy

**Comment by speth** [comment]
Given what Modest has to say about Tien's values, "these values are today known to be in serious error", I think the fits to the RADCAL values from Sandia would be an improvement. Those fits also have the benefit of having better behavior than the Cosilab fit in the 2000-3000 K range, even if they were only generated for T < 2500 K.

**Comment by cannondale1492** [comment]
I integrated the Sandia polynomials with the RADCAL data.

**Comment by speth** [comment]
OK, I think this is basically good to go. Since we're still using Google Code's SVN repo as the canonical source, I'll need to make this into a patch and commit it to the SVN repo. How would you like your name to appear in the acknowledgement in the commit message?

**Comment by cannondale1492** [comment]
Great, please mention me as: Andreas Rücker, Lehrstuhl für Thermodynamik, TU München

**Comment by speth** [comment]
Added to trunk in r3369: https://code.google.com/p/cantera/source/detail?r=3369

Thanks again for the contribution.

**Pull request closed without merging**